packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/ExpoHandlingDelegate.kt

```diff
@@ -56,12 +56,24 @@ class ExpoHandlingDelegate(protected val context: Context) : HandlingDelegate {
 
   fun getListeners() = sListenersReferences.values.mapNotNull { it.get() }
 
+  /**
+   * If the app is foregrounded, invokes [NotificationManager.onNotificationReceived] with the
+   * [notification].
+   *
+   * If the app is backgrounded, a notification is only presented if
+   * the title and text are both non-empty. Otherwise, this is a "data-only" or "silent"
+   * notification that should not be presented to the user.
+   */
   override fun handleNotification(notification: Notification) {
     if (isAppInForeground()) {
       getListeners().forEach {
         it.onNotificationReceived(notification)
       }
-    } else {
+    } else if (!(
+      notification.notificationRequest.content.title.isNullOrEmpty() &&
+        notification.notificationRequest.content.text.isNullOrEmpty()
+      )
+    ) {
```

Comment:
```suggestion
    } else if (shouldPresentNotification(notification)) {
```

we could try to refactor these logic into separated method and the code would also be clear.

```kotlin
/**
 * If the app is backgrounded, a notification is only presented if
 * the title and text are both non-empty. Otherwise, this is a "data-only" or "silent"
 * notification that should not be presented to the user.
 */
private fun shouldPresentNotification(notification: Notification) {
  val content = notification.notificationRequest.content
  if (content.title.isNullOrEmpty() || content.text.isNullOrEmpty()) {
    return false
  }
  return true
}
```